immunisations/README.md

```diff
@@ -0,0 +1,162 @@
+- doseQuantity
+- actor
+- reaction
+- reported
+- doseStatusReason
+
+<details><summary>Example JSON</summary>
+
+```JSON
+{
+     "resource": {
+         "resourceType": "Immunization",
+         "id": "immunization-id",
+         "meta": {
+             "profile": [
+                 "https://fhir.nhs.uk/STU3/StructureDefinition/CareConnect-GPC-Immunization-1"
+             ]
+         },
+         "extension": [
+             {
+                 "url": "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-VaccinationProcedure-1",
+                 "valueExtension": {
+                     "url": "https://fhir.nhs.uk/STU3/StructureDefinition/Extension-coding-sctdescid",
+                     "valueCodeableConcept": {
+                         "text": "Haemophilus influenzae type B and meningitis C vaccination"
+                     }
+                 }
+             },
+             {
+                 "url": "https://fhir.hl7.org.uk/STU3/StructureDefinition/Extension-CareConnect-DateRecorded-1",
+                 "valueString": "20100113151332"
+             }
+         ],
+         "identifier": [
+             {
+                 "system": "https://PSSAdaptor/2167888433",
+                 "value": "immunization-id"
+             }
+         ],
+         "status": "completed",
+         "notGiven": false,
+         "patient": {
+             "reference": "Patient/c2e046b3-6d29-423a-96af-d58640d65e7e"
+         },
+         "encounter": {
+             "reference": "Encounter/2485BC20-90B4-11EC-B1E5-0800200C9A66"
+         },
+         "date": "2010-01-18T11:41:00+00:00",
+         "primarySource": false,
+         "practitioner": [
+             {
+                 "actor": {
+                     "reference": "Practitioner/9C1610C2-5E48-4ED5-882B-5A4A172AFA35"
+                 }
+             }
+         ],
+         "note": [
+             {
+                 "text": "Primary Source: true Location: EMIS Test Practice Location Manufacturer:\n                                                                    another company Batch: past2003 Expiration: 2003-01-17 Site: Right arm GMS : Not\n                                                                    GMS\n                                                                "
+             },
+             {
+                 "text": "End Date: 2010-01-18T11:41:00+00:00"
+             }
+         ]
+     }
+ },
+```
+</details>
+
+
+## XML HL7 > JSON FHIR
+
+An Immunization is primarily mapped from an Observation Statement.
+
+| Mapped to (XML HL7)                                | Mapped from (JSON FHIR / other source )              |
+|----------------------------------------------------|------------------------------------------------------|
+| id / root                                          | `id`                                                 |
+| code / display                                     | `extension.valueExtension.valueCodeableConcept.text` |

```

Comment:
we need a `code [@code]`,  `code[@codeSystem] ` and `code / originalText` too, I think it might be worth extracting this into it own mapper too as its relatively complex.